lib/ui/window.dart

```diff
@@ -152,6 +152,9 @@ class Locale {
   /// the region subtag should be uppercase.
   const Locale(this._languageCode, [ this._countryCode ]) : assert(_languageCode != null);
 
+  /// Empty locale constant. This is an invalid locale.
+  static const Locale none = const Locale('', '');
```

Comment:
why not null?

Reply:
or just assert that it's never empty -- can we ever have no locale at all?

Reply:
There seems to be a large string of calls that do not expect the returned locale to ever be null. I can look into changing it to do null checking though, since that is probably a more robust design in the long run.

Reply:
This mostly applies to when the locale is accessed before the platform has had a chance to pass it over on startup. The empty locale would just result in the resolution to use the default (first) locale.